File: final_report/out_of_bounds_section_pkg/oob_example3_loop_read.cpp
#include <iostream>
int main() {
    int arr[3] = {1, 2, 3};
    int sum = 0;
    for (int i = 0; i <= 3; ++i) {  // Line 5: DANGEROUS — i==3 reads past end
        sum += arr[i];              // Line 6: UB when i==3
    }
    std::cout << sum << "\n";
    return 0;
}

File: final_report/out_of_bounds_section_pkg/oob_example3_loop_read.rs
fn main() {
    let arr = [1, 2, 3];
    let mut sum = 0;
    for i in 0..=3 {               // Line 4: DANGEROUS — i==3 is out of bounds
        sum += arr[i];             // Line 5: runtime panic when i==3
    }
    println!("{}", sum);
}
